substrate/client/rpc-spec-v2/src/archive/archive.rs

```diff
@@ -185,4 +201,51 @@ where
 			Err(error) => MethodResult::err(error.to_string()),
 		})
 	}
+
+	fn archive_unstable_storage(
+		&self,
+		hash: Block::Hash,
+		items: Vec<PaginatedStorageQuery<String>>,
+		child_trie: Option<String>,
+	) -> RpcResult<ArchiveStorageResult> {
+		let items = items
+			.into_iter()
+			.map(|query| {
+				let key = StorageKey(parse_hex_param(query.key)?);
+				let pagination_start_key = query
+					.pagination_start_key
+					.map(|key| parse_hex_param(key).map(|key| StorageKey(key)))
+					.transpose()?;
+
+				// Paginated start key is only supported
+				if pagination_start_key.is_some() &&
+					(query.query_type != StorageQueryType::DescendantsValues &&
```

Comment:
Can you add a method on `StorageQuery::is_descendant_query instead of having to check the query type here?
It would make code much easier to read.